fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -1069,6 +1069,97 @@ def assert_state(self, state: TrainingState) -> None:
             self.training_state == state
         ), f"expected to be in state {state} but current state is {self.training_state}"
 
+    @staticmethod
+    @contextlib.contextmanager
+    def enable_wrap(**kwargs) -> Iterator[None]:
+        """
+        Context manager to wrap modules in FullyShardedDataParallel.
+        Useful for when you'd like to apply the same parameters to all child modules that you wrap.
+
+            Usage::
+
+                with FSDP.enable_wrap(**fully_sharded_params):
+                    # Wraps layer in FSDP if within context
+                    self.l1 = FSDP.wrap(torch.nn.Linear(5, 5))
+                    # Wraps children modules based on bucket_size
+                    self.l2 = FSDP.auto_wrap(TransformerBlock(), bucket_size=1e8)
+
+        """
+        with FullyShardedAutoWrap(**kwargs):
+            yield
+
+    @staticmethod
+    def wrap(module, **kwargs):
```

Comment:
nit: thoughts on renaming `**kwargs` here to `**overrides`?

Reply:
for line 1103-1104, what will happen if outer wrap and inter wrap both have same override key but different values?